src/js/components/FrameworkConfiguration.js

```diff
@@ -0,0 +1,417 @@
+import Icon from "#SRC/js/components/Icon";
+import Util from "#SRC/js/utils/Util";
+import StringUtil from "#SRC/js/utils/StringUtil";
+import CosmosErrorMessage from "#SRC/js/components/CosmosErrorMessage";
+import RouterUtil from "#SRC/js/utils/RouterUtil";
+import FrameworkConfigurationForm
+  from "#SRC/js/components/FrameworkConfigurationForm";
+
+const METHODS_TO_BIND = [
+  "handleJSONToggle",
+  "handleGoBack",
+  "handleServiceReview",
+  "handleEditConfigurationButtonClick",
+  "handleActiveTabChange",
+  "handleFocusFieldChange",
+  "handleCloseConfirmModal",
+  "handleConfirmGoBack",
+  "onFormDataChange",
+  "onFormErrorChange"
+];
+class FrameworkConfiguration extends Component {
+  constructor(props) {
+    super(props);
+
+    const { activeTab, focusField } = this.getFirstTabAndField();
+
+    this.state = {
+      reviewActive: props.isInitialDeploy,
+      activeTab,
+      focusField,
+      jsonEditorActive: false,
+      isConfirmOpen: false,
+      isOpen: true
+    };
+
+    METHODS_TO_BIND.forEach(method => {
+      this[method] = this[method].bind(this);
+    });
+  }
+
+  onFormErrorChange(formErrors) {
+    this.props.onFormErrorChange(formErrors);
+  }
+
+  handleFocusFieldChange(activeTab, focusField) {
+    if (focusField === this.state.focusField) {
+      return false;
+    }
+
+    this.setState({ focusField, activeTab });
+  }
+
+  handleActiveTabChange(activeTab) {
+    const { currentActiveTab } = this.state;
+    const { packageDetails } = this.props;
+    const schema = packageDetails.getConfig();
+
+    if (activeTab === currentActiveTab) {
+      return false;
+    }
+
+    const focusField = this.getFirstFieldInTab(schema, activeTab);
+
+    this.setState({ activeTab, focusField });
+  }
+
+  handleEditConfigurationButtonClick() {
+    const { activeTab, focusField } = this.getFirstTabAndField();
+
+    this.setState({ reviewActive: false, activeTab, focusField });
+  }
+
+  getHashMapRenderKeys(formData) {
+    if (!Util.isObject(formData)) {
+      return {};
+    }
+
+    let renderKeys = {};
+    Object.keys(formData).forEach(key => {
+      const formattedKey = key
+        .toLowerCase()
+        .split(/_|-/)
+        .map(word => StringUtil.capitalize(word))
+        .join(" ");
+
+      renderKeys[key] = formattedKey;
+      renderKeys = Object.assign(
+        renderKeys,
+        this.getHashMapRenderKeys(formData[key])
+      );
+    });
+
+    return renderKeys;
+  }
+
+  getFirstFieldInTab(schema, tab) {
+    // don't assume schema has two-levels of fields with properties
+    // for autofocus we return a field id name, if no fields return string "no-op"
+    if (
+      schema.properties[tab] &&
+      schema.properties[tab].properties &&
+      Object.keys(schema.properties[tab].properties).length >= 1
+    ) {
+      return Object.keys(schema.properties[tab].properties)[0];
+    }
+
+    return "NO-OP";
+  }
+
+  getFirstTabAndField() {
```

Comment:
🎨  it would be nice to move _functions_ like this one into a util and adjust the implementation to not rely on local state...

Reply:
I'm going to hold off doing this for now, but I understand potential benefit of moving these kind of things out of components into pure helpers. 

For this PR, I don't want to do any other major refactors that will require additional testing for regressions.